### PR TITLE
Idle page send heartbeat (close #146)

### LINKED ISSUE #146: Browser idle page doesn't send heartbeat
BTW, we should merge `src/client/browser/idle-page/idle-page.js` and `src/client/browser/idle-page/index.js`.

### PR DESCRIPTION
**Pull request by helen-dikareva**
/cc @inikulin @AlexanderMoskovkin

### CODE REVIEW

File: src/client/browser/idle-page/index.js [hunk 1]
```diff
@@ -1,5 +1,29 @@
...
+            .checkStatus(statusUrl, window.XMLHttpRequest)
+            .then((cmd) => {
+                if (cmd === COMMAND.idle)
+                    window.setTimeout(() => this._checkStatus(statusUrl), CHECK_STATUS_DELAY);
+            })
+            .catch(() => this.statusIndicator.showDisconnection())
+    }
+}
+
+window.init = function (statusUrl, heartbeatUrl) {
```
Comment (inikulin) [inline src/client/browser/idle-page/index.js]: Why not expose `IdlePage` to the `window` and just call `new IdlePage(...)` in the `index.html`?

File: src/client/browser/idle-page/index.js [hunk 2]
```diff
@@ -1,5 +1,29 @@
...
-window.init = function (statusUrl) {
-    new IdlePage(statusUrl);
+
+const CHECK_STATUS_DELAY = 1000;
+
+
+export default class IdlePage {
+    constructor (statusUrl, heartbeatUrl) {
+        this.statusIndicator = new StatusIndicator();
+        browser.startHeartbeat(heartbeatUrl, window.XMLHttpRequest);
```
Comment (inikulin) [inline src/client/browser/idle-page/index.js]: Separate field definition with the method calls by the blank line

**Comment by inikulin** [comment]
\r-

**Comment by helen-dikareva** [comment]
FPR

File: src/client/browser/idle-page/index.js [hunk 3]
```diff
@@ -1,5 +1,28 @@
...
+
+class IdlePage {
+    constructor (statusUrl, heartbeatUrl) {
+        this.statusIndicator = new StatusIndicator();
+
+        browser.startHeartbeat(heartbeatUrl, window.XMLHttpRequest);
+        this._checkStatus(statusUrl);
+    }
+
+    _checkStatus (statusUrl) {
```
Comment (inikulin) [inline src/client/browser/idle-page/index.js]: Argh, just realized that we can turn `statusUrl`  into the `IdlePage` field and don't pass it around.

**Comment by inikulin** [comment]
In general, looks good. Just single minor stylistic issue left.

**Comment by helen-dikareva** [comment]
FPR

File: src/client/browser/idle-page/index.js [hunk 4]
```diff
@@ -1,5 +1,29 @@
...
+        browser.startHeartbeat(heartbeatUrl, window.XMLHttpRequest);
+        this._checkStatus();
+    }
+
+    _checkStatus () {
+        browser
+            .checkStatus(this.statusUrl, window.XMLHttpRequest)
+            .then((cmd) => {
+                if (cmd === COMMAND.idle)
+                    window.setTimeout(() => this._checkStatus(this.statusUrl), CHECK_STATUS_DELAY);
```
Comment (inikulin) [inline src/client/browser/idle-page/index.js]: Now need for `_checkStatus` parameter here.

**Comment by inikulin** [comment]
\r-

**Comment by inikulin** [comment]
@helen-dikareva It requires rebase now

**Comment by helen-dikareva** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Pull request merged**